jms/runtime/src/main/java/io/quarkus/artemis/jms/runtime/health/ConnectionFactoryHealthCheck.java

```diff
@@ -10,20 +14,41 @@
 import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
 import org.eclipse.microprofile.health.Readiness;
 
+import io.quarkus.arc.Arc;
+import io.quarkus.artemis.core.common.ArtemisUtil;
+import io.quarkus.artemis.core.runtime.health.ArtemisHealthSupport;
+
 @Readiness
 @ApplicationScoped
 public class ConnectionFactoryHealthCheck implements HealthCheck {
 
-    @Inject
-    ConnectionFactory connectionFactory;
+    private final Map<String, ConnectionFactory> connectionFactories = new HashMap<>();
+
+    public ConnectionFactoryHealthCheck(
+            @SuppressWarnings("CdiInjectionPointsInspection") ArtemisHealthSupport support) {
+        Set<String> names = support.getConfiguredNames();
+        Set<String> excludedNames = support.getExcludedNames();
+        for (String name : names) {
+            ConnectionFactory connectionFactory = ArtemisUtil.isDefault(name)
+                    ? Arc.container().instance(ConnectionFactory.class).get()
+                    : Arc.container().instance(ConnectionFactory.class, NamedLiteral.of(name)).get();
+            if (!excludedNames.contains(name) && connectionFactory != null) {
+                connectionFactories.put(name, connectionFactory);
+            }
+        }
+    }
 
     @Override
     public HealthCheckResponse call() {
-        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Artemis JMS health check");
-        try (Connection connection = connectionFactory.createConnection()) {
-            builder.up();
-        } catch (Exception e) {
-            builder.down();
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Artemis JMS health check").up();
+        for (Map.Entry<String, ConnectionFactory> namedConnectionFactory : connectionFactories.entrySet()) {
```

Comment:
`var entry` would be better here,

just as:
`final ConnectionFactory connectionFactory` can be better readable as `var connectionFactory`. `final` on local variables isn't really necessary, but that can be left.

At least I thought we don't support Java 8 anymore.

Also final on local variables isn't really necessary, but you can

Reply:
done